backend/src/utilities/tuple-map.ts

```diff
@@ -0,0 +1,55 @@
+export class TupleMap<Key, Value> {
+  private readonly _map = new Map<string, Value>();
+
+  private getKey(key: Key): string {
+    return `${this.keyToString(key)}`;
+  }
+
+  constructor(private readonly keyToString: (key: Key) => string) {}
```

Comment:
You've documented everything except the specificity 😅
It might be worth mentioning what the expectations are on `keyToString`